### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by atok624**
> # Hi Reviewer 👋,
> In this branch I fulfilled:
> 
> * Install CanCanCan in your project.
> * Add a role column to the users table. Remember to use a migration for this.
> * A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
>   
>   * For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.A
> * A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
>   
>   * For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.A
> 
> # ✋

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,30 +1,43 @@
...
 
+  def include_user
+    @posts = @user.posts.includes(:comments)
+  end
+
+  def destroy
+    @post.destroy
+    @post.likes.destroy_all
+    redirect_to user_posts_path(@user)
+  end
```
Comment (Qoosim) [inline app/controllers/posts_controller.rb]: - [ ] Great job implementing authorization rules. However, there's little bug here. I am unable to delete a post with comments. Below is the error I got while trying to do just that. Kindly look into it and fix.

### Error
destroy-error

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (atok624) [reply app/controllers/posts_controller.rb]: We have resolved that
Thank you

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @atok624 and @Abe1able , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Required change by previous reviewer implemented :+1: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by atok624** [comment]
Thanks for the review @t-yanick

**Pull request merged**